relays/ethereum/src/ethereum_deploy_contract.rs

```diff
@@ -102,11 +100,11 @@ pub fn run(params: EthereumDeployContractParams) {
 
 /// Prepare initial header.
 async fn prepare_initial_header(
-	sub_client: &SubstrateRpcClient,
+	sub_client: &SubstrateClient<Rialto>,
 	sub_initial_header: Option<Vec<u8>>,
-) -> Result<(SubstrateHeaderId, Vec<u8>), String> {
+) -> Result<(RialtoHeaderId, Vec<u8>), String> {
 	match sub_initial_header {
-		Some(raw_initial_header) => match SubstrateHeader::decode(&mut &raw_initial_header[..]) {
+		Some(raw_initial_header) => match rialto_runtime::Header::decode(&mut &raw_initial_header[..]) {
```

Comment:
Why not keep all the types in a single type (I'm thinking `Rialto` which implements `Chain`)? That would simplfy migration to a fully generic code in the future.

Reply:
Most of required types are already there. This PR isn't about making existing relay generic - it is about making some RPC-related code generic so that we may reuse the same code in sub2sub. We have separate issue #61 for making this relay more or less generic.